src/js/components/stock-movement-wizard/AddItemsPage.jsx

```diff
@@ -242,19 +291,25 @@ class AddItemsPage extends Component {
       this.props.goToPage(5);
     } else {
       this.props.showSpinner();
-      this.createRequisitionItems(lineItems)
+      this.saveRequisitionItems(lineItems)
         .then(() => {
-          this.updateRequisitionItems(lineItems)
-            .then(() => { this.props.hideSpinner(); this.props.onSubmit(); })
-            .catch(() => this.props.hideSpinner());
+          if (this.state.statusCode === 'CREATED' || this.state.statusCode === 'EDITING') {
+            this.transitionToStep3()
+              .then(() => {
+                this.props.onSubmit();
+              })
+              .catch(() => this.props.hideSpinner());
+          } else {
+            this.props.onSubmit();
+          }
         })
         .catch(() => this.props.hideSpinner());
     }
   }
 
-  createRequisitionItems(lineItems) {
-    const lineItemsToBeAdded = _.filter(lineItems, item => !item.statusCode);
-    const addItemsUrl = `/openboxes/api/stockMovements/${this.props.stockMovementId}`;
+  saveRequisitionItems(lineItems) {
+    const itemsToSave = this.getLineItemsToBeSaved(lineItems);
+    const updateItemsUrl = `/openboxes/api/stockMovements/${this.props.stockMovementId}`;
     const payload = {
       id: this.props.stockMovementId,
```

Comment:
As you mentioned the other day, the stock movement attributes are no longer necessary with these requests. You're ok to keep them in there, but just wanted to point out places to make the code easier to manage for you.

Reply:
Yeah, i just forgot to remove those from requests that were already implemented. I can do some cleanup in payloads once we will have all steps covered.